### PR TITLE
Feat.iframe recipient details AB#16287

### PR DESCRIPTION
**Pull request by PeterSmallenbroek**

### CODE REVIEW

File: interfaces/HO-Portal/src/app/iframe/recipient.page.html [hunk 1]
```diff
@@ -1,5 +1,39 @@
...
+  <ng-container *ngIf="!queryParamPhonenumber">
+    <div class="ion-padding">
+      <h4>{{"page.iframe.recipient.no-phone-number" | translate}}</h4>
+    </div>
+  </ng-container>
+  <ng-container *ngIf="recipients && programsMap">
+    <ion-accordion-group [value]="accordionGroupValue">
+      <ion-accordion
+        *ngFor="let recipient of recipients"
+        value="{{recipient.id}}"
```
Comment (elwinschmitz) [inline interfaces/HO-Portal/src/app/iframe/recipient.page.html:16]:
```suggestion
        [value]="recipient.id"
```

👆 Would be the correct syntax.

I also think that Prettier has to take a good look at this file. ;)

File: interfaces/HO-Portal/src/app/iframe/recipient.page.spec.ts [hunk 2]
```diff
@@ -9,7 +13,13 @@ describe('Recipient.PageComponent', () => {
   beforeEach(waitForAsync(() => {
     TestBed.configureTestingModule({
       declarations: [RecipientPage],
-      imports: [IonicModule.forRoot()],
+      imports: [
+        IonicModule.forRoot(),
+        RouterTestingModule,
+        TranslateModule.forRoot(),
+        HttpClientModule,
```
Comment (elwinschmitz) [inline interfaces/HO-Portal/src/app/iframe/recipient.page.spec.ts:20]:
```suggestion
        HttpClientTestingModule,
```
I haven't checked the exact name, but the point is that the tests shouldn't make REAL HTTP-requests, but fake/mocked ones only.

File: interfaces/HO-Portal/src/app/iframe/recipient.page.ts [hunk 3]
```diff
@@ -1,51 +1,61 @@
...
+  constructor(
+    private activatedRoute: ActivatedRoute,
+    private progamsServiceApiService: ProgramsServiceApiService,
+  ) {
+    this.paramsSubscription = this.activatedRoute.queryParams.subscribe(
+      (params: Params) => {
+        if (!params.phonenumber) {
+          return;
+        }
+        this.queryParamPhonenumber = params.phonenumber;
```
Comment (elwinschmitz) [inline interfaces/HO-Portal/src/app/iframe/recipient.page.ts:29]: This way, when the query-parameter gets 'emptied' or removed (by Twilio Flex), that won't be reflected in the local/public property `this.queryParamPhonenumber`...

File: interfaces/HO-Portal/src/app/components/recipient-details/recipient-details.component.html [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        size="4"
+        *ngFor="let column of orderedColumns"
+      >
+        <ion-item>
+          <h3>{{ translationPrefix + column | translate }}</h3>
+        </ion-item>
+        <ion-item
+          *ngFor="let attr of columns[column]"
+          lines="none"
+          style="--min-height: 30px"
```
[the remark this answers is not in the document]
Comment (arsforza) [reply interfaces/HO-Portal/src/app/components/recipient-details/recipient-details.component.html:14]: done

File: interfaces/HO-Portal/src/app/components/recipient-details/recipient-details.component.html [hunk 5]
```diff
@@ -2,25 +2,48 @@
...
+            lines="none"
+            style="--min-height: 2rem"
+            (click)="buttonClick(recipient, program, transaction)"
+          >
+            <ion-label
+              color="secondary"
+              class="ion-no-margin"
+              style="text-decoration: underline"
+            >
+              {{ 'Payment #' + transaction.payment }}:
```
Comment (elwinschmitz) [inline interfaces/HO-Portal/src/app/components/recipient-details/recipient-details.component.html:39]: Needs a translation-key-string-thing.

**Pull request merged**